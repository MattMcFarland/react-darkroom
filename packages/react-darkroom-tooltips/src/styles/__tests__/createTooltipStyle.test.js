import {
  createTooltipStyle,
} from '../index';

const { describe, test, expect } = global;

describe('styles/createTooltipStyle', () => {
  test('createTooltipStyle creates tooltip stylesheet from given params', () => {
    expect(createTooltipStyle('s', 'color', 'arrowSize', 'radius')).toMatchSnapshot();
  });
});
