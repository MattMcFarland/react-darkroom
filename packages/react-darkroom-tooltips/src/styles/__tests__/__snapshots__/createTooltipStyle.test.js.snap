// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles/createTooltipStyle createTooltipStyle creates tooltip stylesheet from given params 1`] = `
Object {
  "&:after": Object {
    "backgroundColor": "color",
    "borderRadius": "radius",
    "boxShadow": "1px 1px 3px rgba(0, 0, 0, .2)",
    "color": "white",
    "content": "attr(data-label)",
    "padding": "5px 10px",
    "textShadow": "0 -1px 0 rgba(0, 0, 0, .2)",
    "top": "135%",
    "whiteSpace": "nowrap",
  },
  "&:before": Object {
    "border": "arrowSize solid transparent",
    "borderBottomColor": "color",
    "bottom": "-35%",
    "content": "\\"\\"",
    "zIndex": 1000001,
  },
  "&:before, &:after": Object {
    "left": "50%",
    "opacity": 0,
    "pointerEvents": "none",
    "position": "absolute",
    "transform": "translateX(-50%)",
    "transition": "250ms ease-in-out",
    "visibility": "hidden",
    "zIndex": 1000000,
  },
  "&:hover, &:focus": Object {
    "&:after": Object {
      "top": "115%",
    },
    "&:before": Object {
      "bottom": "-15%",
    },
    "&:before, &:after": Object {
      "opacity": 1,
      "visibility": "visible",
    },
  },
  "display": "inline-block",
  "position": "relative",
}
`;
